LiveSPICEVst/SimulationProcessor.cs

```diff
@@ -0,0 +1,322 @@
+using ComputerAlgebra;
+using SharpSoundDevice;
+using Util;
+
+namespace LiveSPICEVst
+{
+    public class ComponentWrapper
+    {
+        public string Name { get; set; }
+        public bool NeedUpdate { get; set; }
+        public bool NeedRebuild { get; set; }
+    }
+
+    /// <summary>
+    /// Simple wrapper around IPotControl to make UI integration easier
+    /// </summary>
+    public class PotWrapper : ComponentWrapper
+    {
+        IPotControl pot = null;
+
+        public PotWrapper(IPotControl pot, string name)
+        {
+            this.pot = pot;
+            this.Name = name;
+        }
+
+        public double PotValue
+        {
+            get
+            {
+                return pot.PotValue;
+            }
+
+            set
+            {
+                pot.PotValue = value;
+
+                NeedUpdate = true;
+            }
+        }
+    }
+
+    /// <summary>
+    /// Simple wrapper around IButtonControl to make UI integration easier
+    /// </summary>
+    public class ButtonWrapper : ComponentWrapper
+    {
+        List<IButtonControl> buttons = new List<IButtonControl>();
+        bool engaged = false;
+
+        public ButtonWrapper(string name)
+        {
+            this.Name = name;
+        }
+
+        public bool Engaged
+        {
+            get
+            {
+                return engaged;
+            }
+
+            set
+            {
+                if (value != engaged)
+                {
+                    engaged = !engaged;
+
+                    foreach (IButtonControl button in buttons)
+                    {
+                        button.Click();
+                    }
+
+                    NeedRebuild = true;
+                }
+            }
+        }
+
+        public void AddButton(IButtonControl button)
+        {
+            buttons.Add(button);
+        }
+    }
+
+    /// <summary>
+    /// Manages single-channel audio circuit simulation
+    /// </summary>
+    public class SimulationProcessor
+    {
+        public ObservableCollection<ComponentWrapper> InteractiveComponents { get; private set; }
+
+        public Schematic Schematic { get; private set; }
+        public string SchematicPath { get; private set; }
+        public string SchematicName { get; private set; }
+
+        public double SampleRate
+        {
+            get { return sampleRate; }
+            set
+            {
+                sampleRate = value;
+
+                needRebuild = true;
+            }
+        }
+
+        public int Oversample
+        {
+            get { return oversample; }
+            set
+            {
+                oversample = value;
+
+                needRebuild = true;
+            }
+        }
+
+        public int Iterations
+        {
+            get { return iterations; }
+            set
+            {
+                iterations = value;
+                needRebuild = true;
+            }
+        }
+
+        double sampleRate = 44100;
```

Comment:
The SimulationProcessor class currently has no dependency on VST plugin stuff, and could potentially be used more generally, so I think a sensible default is probably a good idea. The VST plugin sets the sample rate on Start(), which is always called before any call to ProcessSample(), so it will always be set.